lib/cuda/cudaOutputData.cpp

```diff
@@ -103,17 +123,32 @@ void cudaOutputData::finalize_frame(int frame_id) {
     cudaCommand::finalize_frame(frame_id);
 
     if (!got_metadata) {
-        DEBUG("Passing metadata from input {:s}[{:d}] to {:s}[{:d}]", in_buffer->buffer_name,
-              in_buffer_id, output_buffer->buffer_name, output_buffer_id);
-        pass_metadata(in_buffer, in_buffer_id, output_buffer, output_buffer_id);
-        got_metadata = true;
+        if (in_buffer) {
+            DEBUG("Passing metadata from input (host) buffer {:s}[{:d}] to {:s}[{:d}]",
+                  in_buffer->buffer_name, in_buffer_id, output_buffer->buffer_name,
+                  output_buffer_id);
+            pass_metadata(in_buffer, in_buffer_id, output_buffer, output_buffer_id);
+            got_metadata = true;
```

Comment:
As far as I can tell this variable, `got_metadata`, is not needed.

Reply:
Good catch, that was vestigal